tools/runbook-sync-downstream/main.go

```diff
@@ -0,0 +1,282 @@
+			"CNV runbook '%s' was updated in upstream https://%s at %s.\n"+
+			"This PR syncs the runbook in this repository to contain all new added changes.\n\n"+
+			"/cc @machadovilaca",
+		rb.name, upstreamRepositoryURL, rb.upstreamLastUpdated,
+	)
+
+	err = rbSync.createPR(branchName, commitMessage, body)
+	if err != nil {
+		klog.Fatalf("failed to create PR: %v", err)
+	}
+
+	return branchName
+}
+
+func (rbSync *runbookSync) deleteRunbook(rb runbook) string {
+	runbookName := strings.Replace(rb.name, ".md", "", -1)
+	branchName := fmt.Sprintf("cnv-runbook-delete-%s", runbookName)
+
+	prExists, pr, err := rbSync.prForBranchPreviouslyCreated(branchName)
+	if err != nil {
+		klog.Fatalf("failed to check if branch exists: %v", err)
+	}
+
+	if prExists {
+		klog.Infof("PR for '%s' was previously created: %s", branchName, pr.GetHTMLURL())
+		return branchName
+	}
+
+	worktree, err := newBranchFromMain(rbSync.downstreamRepo, branchName)
+	if err != nil {
+		klog.Fatalf("failed to create branch: %v", err)
+	}
+
+	klog.Infof("deleting file %s", rb.name)
+	err = os.Remove(path.Join(downstreamCloneDir, downstreamRunbooksDir, rb.name))
+	if err != nil {
+		klog.Fatalf("failed to delete file: %v", err)
+	}
+
+	commitMessage := fmt.Sprintf("Delete CNV runbook %s", rb.name)
+
+	err = rbSync.commitAndPush(worktree, commitMessage)
+	if err != nil {
+		klog.Fatalf("failed to push changes: %v", err)
+	}
+
+	body := fmt.Sprintf(
+		"This is an automated PR by 'tools/openshift-virtualization-operator/runbook-sync'.\n\n"+
+			"CNV runbook '%s' was deleted in upstream https://%s.\n"+
+			"This PR deletes the runbook in this repository.\n\n"+
+			"/cc @machadovilaca",
+		rb.name, upstreamRepositoryURL,
+	)
+
+	err = rbSync.createPR(branchName, commitMessage, body)
+	if err != nil {
+		klog.Fatalf("failed to create PR: %v", err)
+	}
+
+	return branchName
+}
+
+func (rbSync *runbookSync) commitAndPush(worktree *git.Worktree, msg string) error {
+	_, err := worktree.Add(downstreamRunbooksDir)
+	if err != nil {
+		return fmt.Errorf("failed to add changes: %w", err)
+	}
+
+	_, err = worktree.Commit(msg, &git.CommitOptions{
+		Author: &object.Signature{
+			Name:  githubUsername,
+			Email: githubEmail,
+			When:  time.Now(),
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to commit changes: %w", err)
+	}
+	klog.Infof("successfully committed: %s", msg)
+
+	if rbSync.dryRun {
+		klog.Warning("[DRY RUN] skipping push")
+		return nil
+	}
+
+	err = rbSync.downstreamRepo.Push(&git.PushOptions{
+		RemoteName: customRemoteName,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to push changes: %w", err)
+	}
+	klog.Info("successfully pushed changes")
+
+	return nil
+}
+
+func (rbSync *runbookSync) prForBranchPreviouslyCreated(branchName string) (bool, *github.PullRequest, error) {
+	prs, _, err := rbSync.ghClient.PullRequests.List(context.Background(), downstreamRepositoryOwner, downstreamRepositoryName, &github.PullRequestListOptions{
+		State: "all",
```

Comment:
Just for ensuring the tool doesn't have bugs, I would change it to `"open|closed"`. IIUC, merged PRs should be taken into account at this point already, as they are part of `master` branch. Their runbooks shouldn't be in the runbooks to change list.